### DISCUSSION TITLE
How can I Declare CircularBuffer variable first and initialize later?

### DISCUSSION THREAD

**Discussion by chenscottus** [rlogiacco/CircularBuffer#77] [General]
How can I Declare  variable first and initialize later?
like this:
      CircularBuffer cb_variable;

      cb_variable = CircularBuffer<int, 100>;

How can I Declare a CircularBuffer pointer and point to a CircularBuffer object?
      CircularBuffer * cb_variable_ptr;

     cb_variable_ptr = &CircularBuffer<int, 100>;

Many thanks!

**Comment by juuldiksWF** [comment]
I'm currently facing the same issue. Facing it when I try to declare the CircularBuffer as member variable in a class. It seems to call the copy constructor (which is deleted) upon initialization of the class that has the CircularBuffer as member.

Any help would be much appriciated!

**Comment by rlogiacco** [comment]
I moved your request to the discussions section, hopefully, some other users might help you. Personally I created and use this structure in a microcontroller context, thus I haven't ever faced this issue.

**Comment by rlogiacco** [comment]
I believe I have an answer for this request: the CircularBuffer is allocated on the stack, not on the heap, and this is more than probably the culprit for your issues.

There is a [closed pull request](https://github.com/rlogiacco/CircularBuffer/pull/80) I invite you to test: if that represents a solution, I will happily merge the PR.
